index.js

```diff
@@ -97,25 +93,19 @@ var EventEmitter = {
      * @param {Function} [fn]
      */
     off: function (name, fn) {
-        if (!isArray(this._listeners)) {
-            this._listeners = [];
-        }
-
-        forEach(this._listeners, function (listener, index) {
-            if (name) {
-                if (listener.name === name) {
-                    if (isFunction(fn)) {
-                        if (listener.fn === fn) {
-                            this._listeners.splice(index, 1);
-                        }
-                    } else {
-                        this._listeners.splice(index, 1);
-                    }
-                }
+        this._listeners = !name ? [] : this._listeners.filter(function (listener, index) {
```

Comment:
I've assumed the browser support from the `forEach` helper:
```
function forEach(list, iterator, context) {
    context = context || {};
    list = list || [];
    Array.prototype.forEach.call(list, iterator.bind(context));
}
```
It uses array's `forEach` method, which has the [same support as the `filter`](http://kangax.github.io/compat-table/es5/) method. I'll add the helper for filter in a moment, but while I'm at it, should I also add a `forEach` polyfill?